core/src/main/java/org/frankframework/pipes/JsonPipe.java

```diff
@@ -71,6 +77,11 @@ public enum Direction {
 	public void configure() throws ConfigurationException {
 		super.configure();
 
+		// rootElementName has been modified from its default value
+		if (getDirection() == Direction.XML2JSON && !DEFAULT_ROOT_ELEMENT_NAME.equals(rootElementName)) {
+			throw new ConfigurationException("rootElementName can not be used when direction is XML2JSON");
+		}
+
```

Comment:
I see that you addressed the Sonar complaint but now, in theory, somebody could manually configure `"root"` as the root element name.

Perhaps you can leave the field `null` by default and then you can check here for field being not null when direction is `XML2JSON`.
And if the direction is `JSON2XML` and it is `null` then set the default?
That should catch all cases.

Reply:
Agreed! Will do